src/poetry/inspection/lazy_wheel.py

```diff
@@ -85,6 +89,24 @@ def metadata_from_wheel_url(
         # themselves are invalid, not because we've messed up our bookkeeping
         # and produced an invalid file.
         raise InvalidWheel(url, name)
+    except Exception as e:
+        if isinstance(e, LazyWheelUnsupportedError):
+            # this is expected when the code handles issues with lazy wheel metadata retrieval correctly
+            raise e
+
+        logger.debug(
```

Comment:
Since you already mentioned it in the chat:

> It is a bit unfortunate that we will end up not really getting reports of lazy wheel not working if there is a bug

We could make this a warning to increase visibility (of course only for unexpected exceptions, see other comment). What do you think?

Reply:
I do not think that adds anything as it will become background noise for most people dealing with this issue as they do not control the ability to manage the index server. It will become an annoying nag rather than serve the purpose of improving their experience.

We can think about how best to balance these concerns in a separate PR I reckon. Might need an option to hide the warning etc.